api/src/main/java/jakarta/data/metamodel/Attribute.java

```diff
@@ -25,111 +25,93 @@
-     */
-    public final Sort asc() {
-        return attrInfo().asc();
-    }
-
+public interface Attribute {
     /**
-     * Obtain a request for an ascending, case insensitive {@link Sort} based on the entity attribute.
+     * Obtain the entity attribute name, suitable for use wherever the specification requires
+     * an entity attribute name. For example, as the parameter to {@link Sort#asc(String)}.
      *
-     * @return a request for an ascending, case insensitive sort on the entity attribute.
+     * @return the entity attribute name.
      * @throws MappingException if an entity attribute with this name does not exist or
      *                          if no Jakarta Data provider provides a repository for the entity type.
      */
-    public final Sort ascIgnoreCase() {
-        return attrInfo().ascIgnoreCase();
-    }
-
-    private final AttributeInfo attrInfo() {
-        AttributeInfo attrInfo = info.get();
-        if (attrInfo == null)
-            throw new MappingException("Unable to find a Jakarta Data provider that provides a repository " +
-                                       " for an entity with an attribute having this name.");
-        return attrInfo;
-    }
+    String name();
 
     /**
-     * Obtain a request for a descending {@link Sort} based on the entity attribute.
+     * <p>Used by the Jakarta Data provider to initialize this {@code Attribute} with implementation.
+     * When initializing a subtype of {@code Attribute} with this method, the Jakarta Data provider
+     * must supply an instance of the subtype.</p>
      *
-     * @return a request for a descending sort on the entity attribute.
-     * @throws MappingException if an entity attribute with this name does not exist or
-     *                          if no Jakarta Data provider provides a repository for the entity type.
-     */
-    public final Sort desc() {
-        return attrInfo().desc();
-    }
-
-    /**
-     * Obtain a request for a descending, case insensitive {@link Sort} based on the entity attribute.
+     * <p>The default value of <code>false</code> is appropriate for instances that are
+     * pre-initialized, such as those that are provided by a Jakarta Data provider
+     * or are generated in advance with an annotation processor.</p>
      *
-     * @return a request for a descending, case insensitive sort on the entity attribute.
-     * @throws MappingException if an entity attribute with this name does not exist or
-     *                          if no Jakarta Data provider provides a repository for the entity type.
+     * @param attr attribute implementation provided by the Jakarta Data provider.
+     * @return true if initialization was successful; false if already initialized.
      */
-    public final Sort descIgnoreCase() {
-        return attrInfo().descIgnoreCase();
+    default boolean init(Attribute attr) {
```

Comment:
IMHO: This method is an encapsulation, that lets the implementation define they way to initialize such as runtime or build time.

Reply:
> IMHO: This method is an encapsulation issue; letting the implementation define their way to initialize, such as runtime or build time sounds better.

I don't see how I introduced any encapsulation issue here.

Before change this code was:
```    public final boolean init(AttributeInfo attrInfo)```

and now it is:
```   default boolean init(Attribute attr)```

`public` is implicit as an interface method, so there is no change there.  The changes that I made to the signature are switching the type of the parameter and making the method `default` so that the Jakarta Data provider doesn't need to bother implementing it when the attribute doesn't require initialization at run time.